src/markdownParser.ts

```diff
@@ -38,7 +38,7 @@ interface ICodeBlockStart {
  * between the start and end blocks, etc. This is good enough for typical use cases.
  */
 function parseCodeBlockStart(line: string): ICodeBlockStart | null {
-	const match = line.match(/(    |\t)?```(\S*)/);
+	const match = line?.match(/(    |\t)?```(\S*)/);
```

Comment:
The signature on this method says line is not null, so either the signature should be updated too, or preferably, we wouldn't call this method when we know that line is null

Reply:
@roblourens I'm not sure how best to fix the real issue. When `parseMarkdown` is called with an empty string, `parseWhitespaceLines` looks for a non-empty line and can't find one, so it sets `i = lines.length` (effectively). Subsequently, `parseMarkdown` calls `parseCodeBlockStart(lines[i])`, but `i == 1` so `typeof lines[i] == 'undefined'`, hence the bug. One solution would be to add an explicit check and return for `content == ''`. However, I worry that might mask other bugs. Without really digging into the code, I don't fully understand why `parseWhitespaceLines` sets `i = lines.length`, but as far as I can tell, any time that case is hit, `parseCodeBlockStart` will be called with `undefined`. I did push a commit that updates the method signature.

Reply:
It appears that this condition is only hit when `content` is empty, because otherwise any empty lines are captured as trailing whitespace for some block. So either `parseWhitespaceLines` has no effect because `i` points to a line with text, or `parseWhitespaceLines` isn't called because the loop exits.

Reply:
That's fine if you want to do that. But basically after `parseWhitespaceLines` advances `i`, you would just need to check `i < lines.length` again and break there, that's probably what I intended.

Reply:
I removed the changes I had made, added a check on `i` after `leadingWhitespace`, and added a test